### PR TITLE
docs(hub) add hub remove code snippet example

### PR DESCRIPTION
**Pull request by helgabalashova**
_Issue #, if available:_ #8807

_Description of changes:_

Adds code snippet and description for Remove Hub listener event section. Moves Channels section to the beginning of the doc.
Screen Shot 2022-10-21 at 4 59 22 PM

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/utilities/js/hub.mdx [hunk 1]
```diff
@@ -73,20 +86,19 @@ The `event` field is recommended to be a small string without spaces such as `si
...
-* auth
-* api
-* analytics
-* interactions
-* pubsub
-* storage
-* xr
-* datastore
+/* start listening for messages */
+const hubEventListener = Hub.listen(/.*/, (data) => {
```
Comment (haverchuck) [inline src/fragments/lib/utilities/js/hub.mdx:95]: I think `cancelToken` might be a better variable name.

File: src/fragments/lib/utilities/js/hub.mdx [hunk 2]
```diff
@@ -7,6 +7,19 @@ import { Hub } from 'aws-amplify';
 
 ## Working with the API
 
+### Channels
+A channel is a logical group name that you use to organize messages and listen on. These are strings and completely up to you as the developer to define for dispatching or listening. However, while you can dispatch to any channel, ***Amplify protects certain channels*** and will flag a warning as sending unexpected payloads could have undesirable side effects (such as impacting authentication flows). The protected channels are currently:
```
Comment (abdallahshaban557) [inline src/fragments/lib/utilities/js/hub.mdx:11]: Can we change this paragraph slightly like so:

The hub module allows you to see different dispatched events in channels that represent the Amplify packages. You also can set your own custom channels. However, while you can dispatch to any channel, ***Amplify protects certain channels*** and will flag a warning as sending unexpected payloads could have undesirable side effects (such as affecting authentication flows). The protected channels are currently:

**Review by abdallahshaban557 — CHANGES_REQUESTED** [review]
Requested some changes - does my feedback make sense? Am I wording things correctly?

File: src/fragments/lib/utilities/js/hub.mdx (hunk 1 above)
Comment (abdallahshaban557) [reply src/fragments/lib/utilities/js/hub.mdx:95]: I feel like `cancelToken` is a little confusing for me. Can you help me understand a bit why that would be better?

File: src/fragments/lib/utilities/js/hub.mdx (hunk 2 above)
Comment (helgabalashova) [reply src/fragments/lib/utilities/js/hub.mdx:11]: As we discussed in Slack, let's just keep it as it is right now.

File: src/fragments/lib/utilities/js/hub.mdx (hunk 1 above)
Comment (helgabalashova) [reply src/fragments/lib/utilities/js/hub.mdx:95]: @haverchuck were you able to discuss it with @abdallahshaban557?

Comment (haverchuck) [reply src/fragments/lib/utilities/js/hub.mdx:95]: How about `hubListenerCancelToken`?

**Review by abdallahshaban557 — APPROVED** [review]

**Review by haverchuck — APPROVED** [review]

**Review by haverchuck — APPROVED** [review]

**Pull request merged**